packages/roosterjs-editor-api/lib/format/getFormatState.ts

```diff
@@ -42,6 +42,13 @@ export function getElementBasedFormatState(
         ? toArray(table.rows[0].cells).every(cell => getTagOfNode(cell) == 'TH')
         : undefined;
 
+    let dir = undefined;
+    let element = editor.getElementAtCursor();
+    while (!dir && element) {
+        dir = element.getAttribute('dir');

```

Comment:
dir can also be specified using CSS "direction".

So do this in getStyleBasedFormatState is much easier.

Reply:
Thanks for suggestion! Moved and renamed it.